components/style/values/specified/text.rs

```diff
@@ -242,3 +244,124 @@ impl ToComputedValue for TextOverflow {
         }
     }
 }
+
+bitflags! {
+    #[derive(MallocSizeOf, ToComputedValue)]
+    /// Specified keyword values for the text-decoration-line property.
+    pub struct TextDecorationLine: u8 {
+        /// No text decoration line is specified
+        const NONE = 0;
+        /// Underline
+        const UNDERLINE = 0x01;
+        /// Overline
+        const OVERLINE = 0x02;
+        /// Line through
+        const LINE_THROUGH = 0x04;
+        /// Blink
+        const BLINK = 0x08;
+        #[cfg(feature = "gecko")]
+        /// Only set by presentation attributes
+        ///
+        /// Setting this will mean that text-decorations use the color
+        /// specified by `color` in quirks mode.
+        ///
+        /// For example, this gives <a href=foo><font color="red">text</font></a>
+        /// a red text decoration
+        const COLOR_OVERRIDE = 0x10;
+    }
+}
+
+impl TextDecorationLine {
+    #[inline]
+    /// Returns the initial value of text-decoration-line
+    pub fn get_initial_value() -> Self {
+        TextDecorationLine {
+            bits: 0
+        }
+    }
+
+    #[cfg(feature = "servo")]
+    #[inline]
+    /// Custom cascade for the text-decoration-line property in servo
+    pub fn cascade_property_custom(_declaration: &PropertyDeclaration, context: &mut Context) {
+        longhands::_servo_text_decorations_in_effect::derive_from_text_decoration(context);
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl_bitflags_conversions!(TextDecorationLine);
+
+impl Parse for TextDecorationLine {
+    /// none | [ underline || overline || line-through || blink ]
+    fn parse<'i, 't>(
+        _context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<TextDecorationLine, ParseError<'i>> {
+        let mut result = TextDecorationLine::NONE;
+        if input.try(|input| input.expect_ident_matching("none")).is_ok() {
+            return Ok(result)
+        }
+        let mut empty = true;
+
+        loop {
+            let result: Result<_, ParseError> = input.try(|input| {
+                let location = input.current_source_location();
+                match input.expect_ident() {
```

Comment:
Given the only thing we do with the error from here is returning it, this can just be `match input.expect_ident()?`, I think.

And that means that the whole match can become:

```rust
try_match_ident_ignore_ascii_case! { input,
    // ...
}
```